x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTruncProcessorTests.java

```diff
@@ -129,6 +143,91 @@ public void testTruncation() {
         assertEquals("2019-09-04T04:10:37.123456789+10:00",
             toString((ZonedDateTime) new DateTrunc(Source.EMPTY, l("nanoseconds"), dateTime, zoneId)
                 .makePipe().asProcessor().process(null)));
+
+        assertEquals("+2000-0",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("millennia"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2000-0",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("CENTURY"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2010-0",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("decades"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-0",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("years"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-9",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("quarters"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("month"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("weeks"), yearToMonth, null)
```

Comment:
Postgres behaves like this:
```
postgres=# select date_trunc('week', INTERVAL '2019 YEARS');
ERROR:  interval units "week" not supported because months usually have fractional weeks
```
I suggest to throw an exception in the relevant truncate function of the `WEEK` Enum value.